pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -98,13 +105,131 @@ type DockercfgController struct {
 
 	serviceAccountCache      MutationCache
 	serviceAccountController *framework.Controller
+	secretController         *framework.Controller
 
 	queue workqueue.RateLimitingInterface
 
 	// syncHandler does the work. It's factored out for unit testing
 	syncHandler func(serviceKey string) error
 }
 
+func (e *DockercfgController) secretTokenAvailable(_, newObj interface{}) {
+	secret, ok := newObj.(*api.Secret)
+	if !ok {
+		return
+	}
+	if len(secret.Annotations) == 0 {
+		return
+	}
+	serviceAccountName, isTokenSecret := secret.Annotations[api.ServiceAccountNameKey]
+	if !isTokenSecret {
+		return
+	}
+	// This token is not waiting for the dockercfg secret
+	if _, isPending := secret.Annotations[PendingDockercfgNameKey]; !isPending {
+		return
+	}
+	if data, exists := secret.Data[api.ServiceAccountTokenKey]; exists && len(data) > 0 {
```

Comment:
ok, I think I see what you mean... when I create the dockerPullSecret and now observing the "addFunc", we should detect that and call addSecretToServiceAccount.